src/Microsoft.AspNet.Mvc.Core/ViewComponents/ViewComponent.cs

```diff
@@ -99,13 +102,60 @@ public ModelStateDictionary ModelState
         /// Gets or sets the <see cref="ViewContext"/>.
         /// </summary>
         [Activate]
-        public ViewContext ViewContext { get; set; }
+        public ViewContext ViewContext
+        {
+            get
+            {
+                if (_viewContext == null)
+                {
+                    // This should run only for the ViewComponent unit test scenarios
+                    _viewContext = new ViewContext();
+                    _viewContext.ViewData = new ViewDataDictionary(
+                        new EmptyModelMetadataProvider(),
+                        _viewContext.ModelState);
```

Comment:
as you've seen, view component testing needs this.  HTML helpers definitely use `ViewData` and having an instance available does no harm in tag helpers (few MVC tag helpers even indirectly use `ViewData`). so I think this basic initialization should be done in one test-only location.